Lib/Inc/CUartCom.h

```diff
@@ -11,25 +11,39 @@
 #ifndef CUARTCOM_H_
 #define CUARTCOM_H_
 
+#include <queue>
 #include <string>
+#include "CCmdBuffer.h"
+#include "CGpioWrapper.h"
+#include "IComChannel.h"
 #include "usart.h"
 
-#define UART_TIMEOUT 100
-
-class CUartCom
+#define UART_TIMEOUT   100
+#define MAX_QUEUE_SIZE 20
+class CUartCom : public IComChannel
 {
-private:
-    UART_HandleTypeDef *mp_huart;
-    GPIO_TypeDef *m_uart_de_port = nullptr;
-    uint16_t m_uart_de_pin;
-
 public:
-    CUartCom(UART_HandleTypeDef *p_huart);
+    CUartCom(UART_HandleTypeDef *p_huart, const std::string name);
     CUartCom(UART_HandleTypeDef *p_huart,
+             const std::string name,
              GPIO_TypeDef *uart_de_port,
              uint16_t uart_de_pin);
 
-    void sendMessage(const std::string &msg);
+    void startReception();
+    void send(const std::string &msg);
+    bool isCommandAvailable();
+    std::string getCommand();
+    void uartRxHandler(UART_HandleTypeDef *p_huart);
+
+    static CUartCom *sp_UART_1;
```

Comment:
Perhaps this is the place to consider having more than two instances of uart controller. Say, you have this as a static array of pointers and a static counter that counts up.

Reply:
Static array initialisation is a bit tricky. You basically need to explicitly initialise entire array to NULL.
```
CUartCom * CUartCom::sp_uart[] = {NULL, NULL, NULL...};
```